src/lib/mina_base/account.ml

```diff
@@ -578,6 +578,8 @@ let min_balance_at_slot ~global_slot ~cliff_time ~cliff_amount ~vesting_period
     ~vesting_increment ~initial_minimum_balance =
   let open Unsigned in
   if Global_slot.(global_slot < cliff_time) then initial_minimum_balance
+    (* If vesting period is zero then everything vests immediately at the cliff *)
+  else if Global_slot.(equal vesting_period zero) then Balance.zero
```

Comment:
I think this is okay, but can we have @mrmr1993 double check?

Reply:
:+1: I'm happy with this, I don't think it would make sense to do anything else here. Good catch @omerzach.